### PR TITLE
New Indonesian Translation: Sharing State Between Components

### LINKED ISSUE #381: Sharing State Between Components
Page: https://id.react.dev/learn/sharing-state-between-components

Assignee: @yaffalhakim1

### PR DESCRIPTION
**Pull request by yaffalhakim1**
Hi, this PR is for #330 
New translation for page `Sharing State Between Components`

Closes #381

### CODE REVIEW

**Comment by yaffalhakim1** [comment]
Hi mas @resir014 and maintainers. This pr is ready to review. You can check in this commits:

- https://github.com/reactjs/id.react.dev/pull/370/commits/e2f46eba536da9c5afb49d8efc22b70f47ffe275
- https://github.com/reactjs/id.react.dev/pull/370/commits/5b1eb14d92d491dddb1ca205495cfec9bf456a97
thank you!

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:4]:
```suggestion
title: Berbagi State Antar Komponen
---
```

Make sure to not mess around with the other Markdown syntax.

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:15]: According to the [Glossary](https://github.com/reactjs/id.react.dev/wiki/Glosarium) the term `controlled component` shouldn't be translated, but that was in the past. Should we change how we translate this? @mhaidarhanif @r17x @zainfathoni

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:62]:
```suggestion
        Nama berasal dari <span lang="kk-KZ">алма</span>, kata Kazakh untuk "apel" dan sering diterjemahkan sebagai "penuh dengan apel". Faktanya, wilayah sekitar Almaty dipercaya sebagai rumah leluhur apel, dan tanaman liar <i lang="la">Malus sieversii</i> dianggap sebagai kandidat yang mungkin untuk leluhur apel domestik modern.
```

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:79]:
```suggestion
Perhatikan bagaimana menekan tombol satu panel tidak memengaruhi panel lainnya--mereka independen.
```

Pastikan tanda bacanya tidak ada yang berubah.

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:125]:
```suggestion
Sekarang komponen induk `Panel` dapat *mengontrol* `isActive` dengan [mengoper sebagai prop.](/learn/passing-props-to-a-component) Sebaliknya, komponen `Panel` sekarang tidak memiliki *kontrol* atas nilai `isActive` - sekarang terserah komponen induk!
```

Extra line here.

File: src/content/learn/sharing-state-between-components.md
Comment (resir014) [inline src/content/learn/sharing-state-between-components.md:242]:
```suggestion
       Nama berasal dari <span lang="kk-KZ">алма</span>, kata Kazakh untuk "apel" dan sering diterjemahkan sebagai "penuh dengan apel". Faktanya, wilayah sekitar Almaty dipercaya sebagai rumah leluhur apel, dan tanaman liar <i lang="la">Malus sieversii</i> dianggap sebagai kandidat yang mungkin untuk leluhur apel domestik modern.
```

**Review by resir014 — DISMISSED** [review]
First pass of code review. Here are some needed changes! @yaffalhakim1

**Comment by yaffalhakim1** [comment]
> First pass of code review. Here are some needed changes! @yaffalhakim1

alright mas, I will commit the suggestion ASAP while waiting for other maintainers to give their reviews.

File: src/content/learn/sharing-state-between-components.md
Comment (mhaidarhanif) [inline src/content/learn/sharing-state-between-components.md:7]: Indonesian `ke` need to be separated.

```suggestion
Terkadang, Anda ingin 2 komponen selalu berubah secara bersamaan. Untuk melakukannya, hapus *state* dari kedua komponen, pindahkan ke komponen induk terdekat, dan kemudian oper ke komponen tersebut melalui *props*. Ini dikenal sebagai mengangkat *state* ke atas, dan ini adalah salah satu hal yang paling umum yang akan Anda lakukan saat menulis kode React.
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Small fix.

**Review by mhaidarhanif — APPROVED** [review]

**Comment by mhaidarhanif** [comment]
Should be fine but let's wait for the others. @resir014 @r17x @mazipan @zainfathoni

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [reply src/content/learn/sharing-state-between-components.md:14]: Yes, I think we should translate it. "Komponen terkendali" is already explicit enough and easily understood.

Comment (zainfathoni) [reply src/content/learn/sharing-state-between-components.md:14]: I would prefer "tak terkendali" to "tidak terkendali", though. It's more concise and simpler.

Comment (mhaidarhanif) [reply src/content/learn/sharing-state-between-components.md:14]: @resir014 @zainfathoni Okay how about:

- `komponen terkendali (*controlled component*)`
- `komponen tak erkendali (*uncontrolled component*)`

As usual, the English words are introduced once in the beginning. Then after that just use the translated version.

File: src/content/learn/sharing-state-between-components.md
Comment (mhaidarhanif) [inline src/content/learn/sharing-state-between-components.md:13]:
```suggestion
- Bagaimana cara berbagi *state* antar komponen dengan menaikkan *state* ke atas (*lifting state up*)
```

**Review by mhaidarhanif — CHANGES_REQUESTED** [review]
Quick fix.

File: src/content/learn/sharing-state-between-components.md
Comment (yaffalhakim1) [reply src/content/learn/sharing-state-between-components.md:14]: it is better with `tak terkendali` mas

**Comment by yaffalhakim1** [comment]
Good afternoon mas @resir014 mas @mhaidarhanif mas @zainfathoni mas @r17x
ready to re-review here: 
https://github.com/reactjs/id.react.dev/pull/370/commits/8b6957e2499a0c6467aa1fc0238fc5d15aee8a3b

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:123]:
```suggestion
Sekarang komponen induk `Panel` dapat *mengontrol* `isActive` dengan [mengoper sebagai prop.](/learn/passing-props-to-a-component) Sebaliknya, komponen `Panel` sekarang tidak memiliki *kontrol* atas nilai `isActive`--sekarang terserah komponen induk!
```

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:130]:
```suggestion
* `Accordion` *(komponen induk terdekat)*
```

Again, it's not capitalized initially, so let's keep it that way.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:103]:
```suggestion
2. **Oper** data yang dituliskan langsung di dalam kode (*hardcoded*) dari komponen induk.
```

I don't think we should leave the "hardcoded" word out of the translated content because it's part of the vital aspect of this step, and we will eventually replace the hardcoded data with a state in the next step.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:126]:
```suggestion
### Langkah 2: Oper data yang dituliskan langsung di dalam kode dari komponen induk {/*step-2-pass-hardcoded-data-from-the-common-parent*/}
```

Let's preserve the "hardcoded" meaning in the translated content because it's part of the vital aspect of this step, and we will eventually replace the hardcoded data with a state in the next step.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:182]:
```suggestion
Coba ubah nilai `isActive` yang dituliskan langsung di dalam kode komponen `Accordion` dan lihat hasilnya di layar.
```

Let's preserve the "hardcoded" meaning in the translated content because it's part of the vital aspect of this step, and we will eventually replace the hardcoded data with a state in the next step.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:186]:
```suggestion
Memindahkan *state* ke atas seringkali mengubah sifat dari apa yang Anda simpan sebagai *state*. 
```

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:190]:
```suggestion
Dalam contoh ini, Anda ingin mengubah `Accordion` sehingga hanya satu panel yang dapat dibuka pada satu waktu. Ini berarti bahwa komponen induk `Accordion` perlu melacak *panel mana* yang sedang aktif. Alih-alih nilai `boolean`, ia dapat menggunakan angka sebagai indeks `Panel` aktif untuk variabel *state*:
```

Because "it" refers to the parent component, let's personify it.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:198]:
```suggestion
Menekan tombol "Tampilkan" di salah satu `Panel` perlu mengubah indeks aktif di `Accordion`. Sebuah `Panel` tidak dapat mengatur *state* `activeIndex` secara langsung karena ia didefinisikan di dalam `Accordion`. Komponen `Accordion` perlu *secara ekplisit mengizinkan* komponen `Panel` untuk mengubah *state*-nya dengan [mengoper *event handler* sebagai prop](/learn/responding-to-events#passing-event-handlers-as-props):
```

- Because "it" refers to the state, let's personify it.
- "explicitly allow" should translate to "*secara eksplisit mengizinkan*".
- make "event handler" italic.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:278]:
```suggestion
Dengan ini, terselesaikanlah pengangkatan *state* ke atas! Memindahkan *state* ke komponen induk memungkinkan Anda untuk mengkoordinasikan kedua panel. Menggunakan indeks aktif alih-alih dua *flag* "ditampilkan" memastikan bahwa hanya satu panel yang aktif pada satu waktu. Dan mengoper *event handler* ke komponen anak memungkinkan komponen anak untuk mengubah *state* induknya.
```

I think it's more accurate to translate "instead of" to "*alih-alih*" rather than *daripada*.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:302]:
```suggestion
Secara umum Anda dapat menyebut sebuah komponen yang memiliki *state* lokal sebagai "tak terkendali". Misalnya, komponen `Panel` yang asli dengan variabel *state* `isActive` adalah tak terkendali karena induknya tidak dapat mempengaruhi apakah panel aktif atau tidak.
```

**Review by r17x — APPROVED** [review]
Need confirm @mhaidarhanif @resir014

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:304]:
```suggestion
Sebaliknya, Anda dapat menyebut sebuah komponen "terkendali" ketika informasi penting di dalamnya dikendalikan oleh *prop* daripada *state* lokalnya sendiri. Ini memungkinkan komponen induk untuk sepenuhnya menentukan perilakunya. Komponen `Panel` akhir dengan *prop* `isActive` dikendalikan oleh komponen `Accordion`.
```

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:306]:
```suggestion
Komponen-komponen tak terkendali lebih mudah digunakan dalam induknya karena membutuhkan konfigurasi yang lebih sedikit. Tetapi mereka kurang fleksibel ketika Anda ingin mengkoordinasikannya bersamaan. Komponen-komponen terkendali sepenuhnya fleksibel, tetapi mereka membutuhkan komponen induk untuk sepenuhnya mengkonfigurasi mereka dengan *props*.
```

Because "they" refers to the components, let's personify it.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:308]:
```suggestion
Pada praktiknya, "terkendali" dan "tak terkendali" bukanlah istilah teknis yang ketat--setiap komponen biasanya memiliki beberapa campuran dari kedua *state* lokal dan *props*. Namun, ini adalah cara yang berguna untuk berbicara tentang bagaimana komponen dirancang dan kemampuan apa yang mereka tawarkan.
```

Adding a "*-lah" particle here emphasizes the meaning of the sentence.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:310]:
```suggestion
Ketika menulis komponen, pertimbangkan informasi manakah di dalamnya yang seharusnya dikendalikan (melalui *props*), dan informasi manakah yang seharusnya tak terkendali (melalui *state*). Tetapi Anda selalu dapat mengubah pikiran Anda dan melakukan refaktorasi belakangan.
```

I think "which" translates to "*manakah*" better than "*apa*". And "should" translates to "*seharusnya*" rather than "*harus*", because "*harus*" is the literal translation for "must".

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:316]:
```suggestion
Pada aplikasi React, banyak komponen akan memiliki *state* mereka sendiri. Beberapa *state* mungkin "hidup" dekat dengan komponen daun (komponen di bagian bawah pohon) seperti masukan-masukan (*inputs*). *State* lainnya mungkin "hidup" lebih dekat ke bagian atas aplikasi. Misalnya, bahkan pustaka-pustaka rute pada sisi klien (*client-side routing libraries*) biasanya diimplementasikan dengan menyimpan rute saat ini di *state* React, dan mengopernya dengan *props*!
```

I think those words need its original English phrases to explain their meaning better.

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:320]:
```suggestion
Aplikasi Anda akan berubah saat Anda mengerjakannya. Biasanya Anda akan memindahkan *state* ke bawah atau kembali ke atas saat Anda masih mencari tahu di mana setiap potongan *state* "hidup". Semua ini adalah bagian dari proses!
```

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Thanks for contributing!
I have a few more suggestions here. 😁

**Comment by yaffalhakim1** [comment]
> Thanks for contributing! I have a few more suggestions here. 😁

surely, I missed a lot 😓, thank you for the review mas. ready to re-review

**Review by mhaidarhanif — APPROVED** [review]

File: src/content/learn/sharing-state-between-components.md
Comment (zainfathoni) [inline src/content/learn/sharing-state-between-components.md:98]:
```suggestion
**Tetapi sekarang katakanlah Anda ingin mengubah panel tersebut sehingga hanya satu panel yang dibuka pada satu waktu.** Dengan desain di atas, membuka panel kedua berarti secara otomatis menutup panel pertama. Bagaimanakah Anda akan melakukannya?
```

I think we can paraphrase it better to make it more understandable.

**Review by zainfathoni — APPROVED** [review]
LGTM 💯 
Approved with a minor suggestion for paraphrasing. 😁 

Thanks for contributing! 🙏

**Pull request merged**
